src/Magento/FunctionalTestingFramework/Util/ModuleResolver.php

```diff
@@ -314,46 +429,193 @@ private static function globRelevantWrapper($testPath, $pattern)
+        ];
+
+        // Add TESTS_MODULE_PATH if it's not included
+        if (array_search($modulePath, $searchCodePaths) === false) {
+            $searchCodePaths[] = $modulePath;
         }
 
-        return $resultArray;
+        return $this->getComposerJsonTestModulePaths($searchCodePaths);
     }
 
     /**
-     * Runs through enabled modules and maps them known module paths by name.
-     * @param array $enabledModules
-     * @param array $allModulePaths
+     * Retrieve all module code paths that have test module composer json files
+     *
+     * @param array $codePaths
      * @return array
      */
-    private function getEnabledDirectoryPaths($enabledModules, $allModulePaths)
+    private function getComposerJsonTestModulePaths($codePaths)
     {
-        $enabledDirectoryPaths = [];
-        foreach ($enabledModules as $magentoModuleName) {
-            if (!isset($this->knownDirectories[$magentoModuleName]) && !isset($allModulePaths[$magentoModuleName])) {
-                continue;
-            } elseif (isset($this->knownDirectories[$magentoModuleName])
-                && !isset($allModulePaths[$magentoModuleName])) {
-                LoggingUtil::getInstance()->getLogger(ModuleResolver::class)->warn(
-                    "Known directory could not match to an existing path.",
-                    ['knownDirectory' => $magentoModuleName]
-                );
+        if (null !== $this->composerJsonModulePaths) {
+            return $this->composerJsonModulePaths;
+        }
+        try {
+            $this->composerJsonModulePaths = [];
+            $resolver = new ComposerModuleResolver();
+            $this->composerJsonModulePaths = $resolver->getTestModulesFromPaths($codePaths);
+        } catch (TestFrameworkException $e) {
+        }
+
+        return $this->composerJsonModulePaths;
+    }
+    
+    /**
+     * Aggregate all code paths with composer installed test modules
+     *
+     * @return array
+     */
+    private function aggregateTestModulePathsFromComposerInstaller()
+    {
+        // Define the module paths
+        $magentoBaseCodePath = MAGENTO_BP;
+        $composerFile = $magentoBaseCodePath . DIRECTORY_SEPARATOR . 'composer.json';
+
+        return $this->getComposerInstalledTestModulePaths($composerFile);
+    }
+
+    /**
+     * Retrieve composer installed test module code paths
+     *
+     * @params string $composerFile
+     * @return array
+     */
+    private function getComposerInstalledTestModulePaths($composerFile)
+    {
+        if (null !== $this->composerInstalledModulePaths) {
+            return $this->composerInstalledModulePaths;
+        }
+        try {
+            $this->composerInstalledModulePaths = [];
+            $resolver = new ComposerModuleResolver();
+            $this->composerInstalledModulePaths = $resolver->getComposerInstalledTestModules($composerFile);
+        } catch (TestFrameworkException $e) {
```

Comment:
Should we at least log these exceptions?

Reply:
This should be logged by ComposerModuleResolver as an exception automatically, is it sufficient?